vizro-core/docs/pages/user-guides/data.md

```diff
@@ -268,8 +268,11 @@ data_manager["no_expire_data"].timeout = 0
 
 ### Parametrize data loading
 
-You can supply arguments to your dynamic data loading function that can be modified from the dashboard.
-For example, if you are handling big data then you can use an argument to specify the number of entries or size of chunk of data.
+You can supply arguments to your dynamic data loading function that can be modified from the dashboard. For example:
+- To load different versions of the same data.
+- To handle big data you can use an argument that controls the amount of data that is loaded. This effectively pre-filters data before it reaches the Vizro dashboard.
```

Comment:
The example below covers the second bullet point. Should we add an example for the first bullet point as well? I thought the code examples would come in the same order as the bullet points 😄

Reply:
Very good idea. Actually I'd never really thought of the example shown covering the second bullet point. It's more that it's the simplest possible example that I could think of, but you're right that a random sampling of points _sort of_ represents pre-filtering the data. The reason I say _sort of_ is because in reality you probably wouldn't want to randomly sample points. Instead the pre-filtering would take in `species` or a number of rows and return something deterministic based on that.

So in the end I think we could do three examples:
- simplest possible just to demonstrate basic principles, like now (if it still seems valuable given the below two)
- loading different versions of data
- handle big data

I'll leave adding any more examples for now and let's revisit later once we've got this feature fully completed since I'd like to spend some time coming up with really good examples here that can demonstrate all the different parts of the feature.

Reply:
Sounds good. Maybe in that case you need to include a sentence to say something like "The following is a simple example to illustrate X" so it's not interpreted as one or the other bullet and also introduces the example clearly.

Reply:
Nice idea, done!